### DISCUSSION TITLE
How do you get access to the cache from an JWTAuthenticationMiddleware?

### DISCUSSION THREAD

**Discussion by danechitoaie** [litestar-org/litestar#1337] [Q&A] [answered]
In the docs I see it being accessed using `scope["app"].cache` but the authenticate_request of JWTAuthenticationMiddleware does not have a scope param?

**Comment by provinzkraut** [comment, marked as answer]
> but the authenticate_request of JWTAuthenticationMiddleware does not have a scope param?

`scope` is an attribute of `ASGIConnection`, which is passed into `authenticate_request`.
So you could access the scope like this: `connection.scope`, and get the cache with `connection.scope["app"].cache`.

You don't need to go through so many indirections though, as `ASGIConenction` directly exposes `app` and `cache` properties, which means the easiest way to access it is `connection.cache`.

Check out the [`ASGIConnection` docs](https://starliteproject.dev/lib/reference/connection/base.html#starlite.connection.base.ASGIConnection).
